zk_toolbox/crates/zk_supervisor/src/commands/test/fund.rs

```diff
@@ -0,0 +1,106 @@
+use std::path::PathBuf;
+
+use anyhow::Context;
+use common::{config::global_config, spinner::Spinner};
+use config::EcosystemConfig;
+use ethers::{
+    signers::{coins_bip39::English, MnemonicBuilder, Signer},
+    types::H160,
+};
+use serde::Deserialize;
+use types::{L1Network, WalletCreation};
+use xshell::Shell;
+
+use crate::messages::{MSG_CHAIN_NOT_FOUND_ERR, MSG_DISTRIBUTING_ETH_SPINNER};
+
+const TEST_WALLETS_PATH: &str = "etc/test_config/constant/eth.json";
+const AMOUNT_FOR_DISTRIBUTION_TO_WALLETS: u128 = 1000000000000000000000;
+
+pub async fn run(shell: &Shell) -> anyhow::Result<()> {
+    let ecosystem = EcosystemConfig::from_file(shell)?;
+
+    let chain = ecosystem
+        .load_chain(global_config().chain_name.clone())
+        .context(MSG_CHAIN_NOT_FOUND_ERR)?;
+
+    if chain.wallet_creation == WalletCreation::Localhost
+        && ecosystem.l1_network == L1Network::Localhost
+    {
+        let spinner = Spinner::new(MSG_DISTRIBUTING_ETH_SPINNER);
+
+        let wallets_path: PathBuf = ecosystem.link_to_code.join(TEST_WALLETS_PATH);
+        let test_wallets: TestWallets =
+            serde_json::from_str(shell.read_file(&wallets_path)?.as_ref())
+                .context("Impossible to deserialize test wallets")?;
+
+        let wallets = ecosystem.get_wallets()?;
+
+        common::ethereum::distribute_eth(
+            wallets.operator,
+            test_wallets.address_list()?,
+            chain
+                .get_secrets_config()?
+                .l1
+                .context("No L1 secrets available")?
+                .l1_rpc_url
+                .expose_str()
+                .to_owned(),
+            ecosystem.l1_network.chain_id(),
+            AMOUNT_FOR_DISTRIBUTION_TO_WALLETS,
+        )
+        .await?;
+
+        spinner.finish();
+    }
+
+    Ok(())
+}
+
+#[derive(Deserialize)]
+struct TestWallets {
+    #[serde(rename = "web3_url")]
+    _web3_url: String,
+    #[serde(rename = "test_mnemonic")]
+    _test_mnemonic: String,
+    test_mnemonic2: String,
+    test_mnemonic3: String,
+    test_mnemonic4: String,
+    test_mnemonic5: String,
+    test_mnemonic6: String,
+    test_mnemonic7: String,
+    test_mnemonic8: String,
+    test_mnemonic9: String,
+    test_mnemonic10: String,
+    #[serde(rename = "mnemonic")]
+    _mnemonic: String,
+    base_path: String,
+}
+
```

Comment:
do you think it's a good approach? 
we don't need to distribute it every time, and in 99% of the cases first N wallets are already prefunded. 
I'd prefer to fund them once right before the test

Reply:
Addressed in https://github.com/matter-labs/zksync-era/pull/2594/commits/9e1e02fac18037ac4d46c156942fb9a269cfe51c